siege_utilities/core/sql_safety.py

```
@@ -40,12 +63,73 @@ def validate_sql_identifier(name: str, label: str = "identifier") -> str:
+    Raises:
+        ValueError: *name* is structurally unsafe OR not in *allowed*.
+    """
+    validate_sql_identifier(name, label, allow_dotted=False)
+    if name not in allowed:
+        sample = sorted(allowed)[:10]
+        more = "..." if len(allowed) > 10 else ""
         raise ValueError(
-            f"Invalid SQL {label}: {name!r} — "
-            f"must match [a-zA-Z_][a-zA-Z0-9_]* (no spaces, dashes, quotes, or special characters)"
+            f"SQL {label} {name!r} not in allowed set ({sample}{more})"
         )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)